api/validation/schemas.py

```diff
@@ -53,72 +51,72 @@ class GetInstanceTypesSchema(Schema):
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
     region = fields.String(required=True, validate=aws_region_validator)
 
 QueueStatus = QueueStatusSchema(unknown=INCLUDE)
 
 
 class ScontrolJobSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    job_id = fields.String(required=True)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
+    job_id = fields.String(required=True, validate=validate.Length(max=256))
     region = fields.String(required=True, validate=aws_region_validator)
 
 ScontrolJob = ScontrolJobSchema(unknown=INCLUDE)
 
 
 class CancelJobSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    job_id = fields.String(required=True)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
+    job_id = fields.String(required=True, validate=validate.Length(max=256))
     region = fields.String(required=True, validate=aws_region_validator)
 
 CancelJob = CancelJobSchema(unknown=INCLUDE)
 
 
 class SacctSchema(Schema):
-    user = fields.String()
-    instance_id = fields.String(required=True)
-    cluster_name = fields.String(required=True, validate=is_alphanumeric_with_hyphen)
+    user = fields.String(validate=validate.Length(max=64))
+    instance_id = fields.String(required=True, validate=validate.Length(max=60))
+    cluster_name = fields.String(required=True, validate=validate.And(is_alphanumeric_with_hyphen, validate.Length(max=60)))
     region = fields.String(required=True, validate=aws_region_validator)
 
 Sacct = SacctSchema(unknown=INCLUDE)
 
 
 class LoginSchema(Schema):
-    code = fields.String(required=True)
+    code = fields.String(required=True, validate=validate.Length(max=128))
 
 Login = LoginSchema(unknown=INCLUDE)
 
 
 class PushLogSchema(Schema):
     class PushLogEntrySchema(Schema):
         level = fields.String(required=True, validate=valid_api_log_levels_predicate)
-        message = fields.String(required=True)
+        message = fields.String(required=True, validate=validate.Length(max=246000)) # CW limit is 256k, leaving 1k to extra and level
         extra = fields.Dict()
 
     logs = fields.List(fields.Nested(PushLogEntrySchema), required=True)
 
 PushLog = PushLogSchema(unknown=INCLUDE)
 
 class PriceEstimateSchema(Schema):
-    cluster_name = fields.String(required=True)
-    queue_name = fields.String(required=True)
+    cluster_name = fields.String(required=True, validate=validate.Length(max=60))
+    queue_name = fields.String(required=True, validate=validate.Length(max=60))
     region = fields.String(validate=aws_region_validator, required=True)
 
 PriceEstimate = PriceEstimateSchema(unknown=INCLUDE)
 
 class PCProxySchema(Schema):
```

Comment:
issue(possibly-blocking): with a schema like that a user is still allowed to use a huge payload proxied to the ParallelCluster API. Since we don't know beforehand the body content of mutations routed to PC, can we at least put a hard limit on the bytes of the body?

Reply:
I will investigate this, I think it's an important issue. Thanks!